activemodel/lib/active_model/type/string.rb

```diff
@@ -11,6 +11,14 @@ def changed_in_place?(raw_old_value, new_value)
         end
       end
 
+      def to_immutable_string
+        ImmutableString.new(
```

Comment:
`ImmutableString` and `MysqlString` are not compatible for boolean serialization.
It seems that `ImmutableMysqlString` is needed.

Reply:
So I think we are missing some test for it.